tests/e2e/configurer/chain/queries.go

```diff
@@ -0,0 +1,150 @@
+package chain
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v7/tests/e2e/util"
+	superfluidtypes "github.com/osmosis-labs/osmosis/v7/x/superfluid/types"
+)
+
+func (c *Config) QueryRPC(path string) ([]byte, error) {
+	var err error
+	var resp *http.Response
+	retriesLeft := 5
+	for {
+		resp, err = http.Get(path)
+
+		if resp.StatusCode == http.StatusServiceUnavailable {
+			retriesLeft--
+			if retriesLeft == 0 {
+				return nil, err
+			}
+			time.Sleep(10 * time.Second)
+		} else {
+			break
+		}
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	bz, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return bz, nil
+}
+
+func (c *Config) QueryChainStatus(validatorIdx int) []byte {
+	cmd := []string{"osmosisd", "status"}
+	_, errBuf, err := c.containerManager.ExecCmd(c.t, c.Id, validatorIdx, cmd, "")
+	require.NoError(c.t, err)
+	return errBuf.Bytes()
+}
+
+func (c *Config) QueryCurrentChainHeightFromValidator(validatorIdx int) int {
+	var block syncInfo
+	require.Eventually(
+		c.t,
+		func() bool {
+			out := c.QueryChainStatus(validatorIdx)
+			err := json.Unmarshal(out, &block)
+			return err == nil
+		},
+		1*time.Minute,
+		time.Second,
+		"Osmosis node failed to retrieve height info",
+	)
+	currentHeight, err := strconv.Atoi(block.SyncInfo.LatestHeight)
+	require.NoError(c.t, err)
+	return currentHeight
+}
+
+func (c *Config) QueryBalances(validatorIndex int, addr string) (sdk.Coins, error) {
+	cmd := []string{"osmosisd", "query", "bank", "balances", addr, "--output=json"}
+	outBuf, _, err := c.containerManager.ExecCmd(c.t, c.Id, validatorIndex, cmd, "")
```

Comment:
same with this, we could probably require "balances:" in the response

Reply:
added